x/ccv/provider/types/keys.go

```diff
@@ -143,30 +143,36 @@ func InitTimeoutTimestampKey(chainID string) []byte {
 	return append([]byte{InitTimeoutTimestampBytePrefix}, []byte(chainID)...)
 }
 
-// PendingCAPKey returns the key under which a pending consumer addition proposal is stored
+// PendingCAPKey returns the key under which a pending consumer addition proposal is stored.
+// The key has the following format: PendingCAPBytePrefix | timestamp | chainID
 func PendingCAPKey(timestamp time.Time, chainID string) []byte {
-	return TsAndChainIdKey(PendingCAPBytePrefix, timestamp, chainID)
-}
-
-// ParsePendingCAPKey returns the time and chain ID for a pending consumer addition proposal key
-// or an error if unparsable
-func ParsePendingCAPKey(bz []byte) (time.Time, string, error) {
-	return ParseTsAndChainIdKey(PendingCAPBytePrefix, bz)
+	timeBz := sdk.FormatTimeBytes(timestamp)
+	return AppendMany(
+		// Append the prefix
+		[]byte{PendingCAPBytePrefix},
+		// Append the time bytes
+		timeBz,
```

Comment:
I don't think this causes any immediate problems, but it does go against the efforts that were made in https://github.com/cosmos/interchain-security/pull/242

It'd be preferable to stay consistent to the decision that was previously made and enforce that any variable length bytes in a key are always at the end of a key. This could be accomplished here by removing chainID from the key and storing an array of props for each timestamp

Reply:
> It'd be preferable to stay consistent to the decision that was previously made and enforce that any variable length bytes in a key are always at the end of a key.

Not sure I understand the reason for this. Also, this is already broken, see `provider/types/keys.go`, e.g., https://github.com/cosmos/interchain-security/blob/3e339eaad34f37f4757049df775ea2c9a8fe941e/x/ccv/provider/types/keys.go#L366

Reply:
Replaced time.Time with an uint64.